CMakeLists.txt

```diff
@@ -8,23 +8,28 @@ project(simdutf
 
 set(SIMDUTF_LIB_VERSION "1.0.1" CACHE STRING "simdutf library version")
 set(SIMDUTF_LIB_SOVERSION "1" CACHE STRING "simdutf library soversion")
+option(SIMDUTF_TESTS "Whether the tests are included as part of the CMake Build." ON)
+option(SIMDUTF_BENCHMARKS "Whether the benchmarks are included as part of the CMake Build." ON)
 
 include(GNUInstallDirs)
 include(cmake/simdutf-flags.cmake)
 
 set(SIMDUTF_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
-enable_testing()
 add_subdirectory(src)
-add_subdirectory(tests)
 add_subdirectory(singleheader)
-
-if(CMAKE_CXX_COMPILER_ID MATCHES GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0)
-  message(STATUS "The benchmark tool requires GCC 8.0 or better.")
-else()
-  add_subdirectory(benchmarks)
+if (SIMDUTF_TESTS)
```

Comment:
There is already a `BUILD_TESTING` variable defined by CTest which you should use instead: https://cmake.org/cmake/help/latest/module/CTest.html

Reply:
If I turn on tests in my branch, but don't want to turn on tests for this, does that mean I need to figure out how to properly directory-scope a change to the `BUILD_TESTING` flag to turn it off for this specifically?

Or, is it maybe enough to change this line:

```cmake
option(SIMDUTF_TESTS "Whether the tests are included as part of the CMake Build." ON)
```

to this:

```cmake
option(SIMDUTF_TESTS "Whether the tests are included as part of the CMake Build." ${BUILD_TESTING})
```

 as a way to "default" to whatever is on for the usage of CTest?

Reply:
No, you should definitely be using `BUILD_TESTING`, as explained by the CTest documentation. Your particular problem with scoped disabling of tests when using FetchContent could be done like this:
```cmake
# BUILD_TESTING is only overwritten in the scope of the function and **not** globally 
function(fetch_simdutf)
  FetchContent_Declare(...)
  set(BUILD_TESTING OFF)
  FetchContent_MakeAvailable(...)
endfunction()

fetch_simdutf()
```
But as @friendlyanon would say, vendoring is almost always a bad decision and you should probably be using the package from your system package manager or vcpkg anyway.

Reply:
I... didn't know using FetchContent was vendoring; in fact, I thought it was the opposite.

But, okay, I can just change it to be `BUILD_TESTING` instead.

Reply:
I changed it to be BUILD_TESTING now.

Reply:
`FetchContent` just does the fetching, `FetchContent_MakeAvailable` will still call `add_subdirectory` if there is a CMakeLists.txt file there, so you effectively own the code, i.e. vendor it.